### ISSUE TITLE
FromBody wouldnt honor System.Text.Json.JsonPropertyName for deserializing

### ISSUE THREAD

**Issue by Jebarson** [dotnet/aspnetcore#40766]
```
 public class Foo
    {
        [JsonPropertyName("Blah")]
        public SingleMailboxNotification[] BlahProp { get; set; }
    }
```

Json Text

`{"Blah" : { "SomeProp" : "SomeVal"} }`

"[FromBody] Foo" will return BlahProp as null. However, if I were to use Newtonsoft.Json.JsonProperty as the attribute, it would deserialize properly.

**Comment by TanayParikh** [comment]
Since `BlahProp` is an array, I believe the JSON should be of the form `{"Blah" : [ { "SomeProp" : "SomeVal"} ] }`.

**Comment by Jebarson** [comment]
Yes sorry that was a typo

**Comment by rafikiassumani-msft** [comment]
@Jebarson Triage: Are you using both `System.Text.Json` and `Newtonsoft.Json` in the same app?

**Comment by Jebarson** [comment]
@rafikiassumani-msft  yes they both are referenced. But this type wasnt using the Newtonsoft

**Comment by brunolins16** [comment]
> @rafikiassumani-msft Rafiki Assumani FTE yes they both are referenced. But this type wasnt using the Newtonsoft

@Jebarson  if you are calling [AddNewtonsoftJson](https://docs.microsoft.com/en-us/aspnet/core/web-api/advanced/formatting?view=aspnetcore-6.0#add-newtonsoftjson-based-json-format-support) in your application startup, that means the `System.Text.Json` formatters were replaced by the `Newtonsoft.Json` formatters, so, the only attributes that will have effect are attributes from `Newtonsoft.Json`.

https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Mvc/Mvc.NewtonsoftJson/src/DependencyInjection/NewtonsoftJsonMvcOptionsSetup.cs#L61

**Comment by Jebarson** [comment]
Let me verify this behavior using a small application and get back to you. Chances are the internal sdk i am using could be doing the same as you mentioned.
